### PR TITLE
IRSA-5099: SHA migration: Add abstract search

### PR DESCRIPTION
**Pull request by jaladh-singhal**
Fixes [IRSA-5099](https://jira.ipac.caltech.edu/browse/IRSA-5099) - linked PR: https://github.com/IPAC-SW/irsa-ife/pull/258

Most of changes are in TablePanel and child components to make it possible to render a data list (single column table)
- Added option to hide header
- Set cell width differently if single column table
- Added option to set row height dynamically (based on cell content)

## Testing
https://irsa-5099-abstract-search.irsakudev.ipac.caltech.edu/applications/Spitzer/SHA - do abstract search, resize screen width, the table content will wrap correctly for tablet widths too.

Do other searches to make sure table panel is rendering as it used to before (to test if it's not broken for more than one column)

### CODE REVIEW

File: src/firefly/js/tables/ui/BasicTableView.jsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const BY_SCROLL = 'byScroll';
 const BasicTableViewInternal = React.memo((props) => {
 
     const {width, height} = props.size;
-    const {columns, data, hlRowIdx, renderers, bgColor, selectInfoCls, callbacks, rowHeight,
+    const {columns, data, hlRowIdx, renderers, bgColor, selectInfoCls, callbacks, rowHeight, rowHeightGetter, showHeader=true,
```
Comment (loitly) [inline src/firefly/js/tables/ui/BasicTableView.jsx:32]: Can you also include the added option `showHeader` to tables-typedefs.jsdoc and table-api.html?  I like to keep them in sync.

File: src/firefly/js/tables/ui/BasicTableView.jsx [hunk 2]
```diff
@@ -75,19 +75,30 @@ const BasicTableViewInternal = React.memo((props) => {
     const adjScrollTop = correctScrollTopIfNeeded(totalColWidths, scrollTop, width, height-headerHeight, rowHeight, hlRowIdx, triggeredBy);
     const adjScrollLeft = correctScrollLeftIfNeeded(totalColWidths, scrollLeft, width, triggeredBy);
 
+    const isSingleColumnTable = (columns) => columns.reduce(
+        (numVisibleCols, col) => get(col, ['visibility'], 'show') === 'show'
+            ? numVisibleCols + 1 : numVisibleCols, 0)
+        === 1;
```
Comment (loitly) [inline src/firefly/js/tables/ui/BasicTableView.jsx:81]: I prefer `filter().length` over `reduce` but, you can decide.  Or, you can use `TableUtil.getColsByType(columns)`.  It abstracts the knowledge of 'visibility', 'show', etc.

**Review by loitly — APPROVED** [review]
Looks good to me.

File: src/firefly/js/tables/ui/BasicTableView.jsx (hunk 2 above)
Comment (jaladh-singhal) [reply src/firefly/js/tables/ui/BasicTableView.jsx:81]: Yeah, filter().length is neater - it didn't occur to me earlier.

**Pull request merged**